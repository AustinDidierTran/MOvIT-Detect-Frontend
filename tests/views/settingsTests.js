/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import Settings from '../../src/views/settings';

Enzyme.configure({ adapter: new Adapter() });

describe('Settings Tests', () => {
  const initialState = {
    applicationReducer: { language: 'en' },
    settingsReducer: {
      totalMemory: 5,
      usedMemory: 10,
    },
  };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {};

  it('should have proptypes', () => {
    const actualValue = Settings.WrappedComponent.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      header: PropTypes.object,
      dataAgreement: PropTypes.bool.isRequired,
      totalMemory: PropTypes.number.isRequired,
      usedMemory: PropTypes.number.isRequired,
      snoozeTime: PropTypes.number.isRequired,
      isLedBlinkingEnabled: PropTypes.bool.isRequired,
      isVibrationEnabled: PropTypes.bool.isRequired,
      changeDataAgreement: PropTypes.bool.isRequired,
      changeTotalMemory: PropTypes.func.isRequired,
      changeUsedMemory: PropTypes.func.isRequired,
      changeSnoozeTime: PropTypes.func.isRequired,
      changeIsLedBlinkingEnabled: PropTypes.func.isRequired,
      changeIsVibrationEnabled: PropTypes.func.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<Settings store={store} {...props} />).dive();
    wrapper.setState({ isLoaded: true });

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
