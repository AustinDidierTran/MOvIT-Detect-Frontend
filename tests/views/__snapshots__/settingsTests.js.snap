// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Tests should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row mt-4"
  >
    <div
      className="col-12"
    >
      <h2
        className="header text-center"
      >
        settings.en
      </h2>
      <Connect(Notification) />
      <CustomCard
        element={<Connect(ModuleStatus) />}
        title="settings.modules.en"
      />
      <CustomCard
        element={<Connect(NotificationSettings) />}
        title="settings.notification.en"
      />
      <CustomCard
        element={<Connect(Permissions) />}
        title="settings.permissions.en"
      />
      <CustomCard
        element={
          <div>
            <h6>
              settings.system.memory.en
            </h6>
            <Connect(MemoryUsage) />
            <br />
            <h6>
              settings.system.update.en
            </h6>
            <Connect(UpdatesManager) />
          </div>
        }
        title="settings.system.en"
      />
    </div>
  </div>
</div>
`;
